pkg/hs/urbit-king/lib/Urbit/Vere/Ames.hs

```diff
@@ -17,15 +17,33 @@ import Urbit.Vere.Ames.DNS (galaxyPort, resolvServ)
 import Urbit.Vere.Ames.UDP (UdpServ(..), fakeUdpServ, realUdpServ)
 
 
+-- Constants -------------------------------------------------------------------
+
+-- | How many unprocessed ames packets to allow in the queue before we start
+-- dropping incoming packets.
+queueBound :: Word
+queueBound = 1000
+
+-- | How often, measured in number of packets dropped, we should announce packet
+-- loss.
+packetsDroppedPerComplaint :: Word
+packetsDroppedPerComplaint = 1000
+
+
 -- Types -----------------------------------------------------------------------
 
 data AmesDrv = AmesDrv
   { aTurfs    :: TVar (Maybe [Turf])
+  , aDropped  :: TVar Word
```

Comment:
I used a TVar to anticipate having it read from another thread (the signal handler), per Joe's remark on SIGINFO.